client/app/lib/redux/modules/bongo.coffee

```diff
@@ -0,0 +1,77 @@
+_ = require 'lodash'
+immutable = require 'app/util/immutable'
+
+{ makeNamespace, expandActionType } = require 'app/redux/helper'
+
+withNamespace = makeNamespace 'koding', 'bongo'
+
+LOAD = expandActionType withNamespace 'LOAD'
+REMOVE = expandActionType withNamespace 'REMOVE'
+
+reducer = (state = immutable({}), action) ->
+
+  switch action.type
+    when LOAD.SUCCESS
+      { result } = action
+      result = [result]  unless Array.isArray result
+      result.forEach (res) ->
+        state = state.set res.constructor.name, immutable {}  unless state[res.constructor.name]
+        state = state.update res.constructor.name, (collection) ->
+          collection = collection.set res._id, immutable res
+          return collection
+
+      return state
+
+    when REMOVE.SUCCESS
+      { result } = action
+      result = [result]  unless Array.isArray result
+      result.forEach (res) ->
+        { _id, constructor: { name: constructorName } } = res
+        state = state.set constructorName, state[constructorName].without(_id)
+
+      return state
+
+    else
+      return state
```

Comment:
remove all `return state` and remove `else` part then just put one `return state` at the end.